plugins/src/test/java/tech/pegasys/pantheon/plugins/TestPlugin.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.plugins;
+
+import tech.pegasys.pantheon.plugins.services.PantheonEvents;
+import tech.pegasys.pantheon.plugins.services.PicoCLIOptions;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.Optional;
+
+import com.google.auto.service.AutoService;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import picocli.CommandLine.Option;
+
+@AutoService(PantheonPlugin.class)
+public class TestPlugin implements PantheonPlugin {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private PantheonContext context;
+  private Optional<Object> listenerReference;
+
+  @Option(names = "--Xtest-option", hidden = true, defaultValue = "UNSET")
+  String testOption;
+
+  @Override
+  public void register(final PantheonContext context) {
+    this.context = context;
+
+    if (System.getProperty("testPlugin.failAtRegister") != null) {
```

Comment:
Any plans to add tests that hit these failure cases or check that the registered CLI flags are used?

Reply:
Yes, with this commit.  Validating the flag is a bit tricky and needs to be done as part of the acceptance tests.